public/expressions/overlay_anomalies.ts

```diff
@@ -0,0 +1,193 @@
+const getAnomalies = async (
+  detectorId: string,
+  startTime: number,
+  endTime: number
+): Promise<AnomalyData[]> => {
+  const anomalySummaryQuery = getAnomalySummaryQuery(
+    startTime,
+    endTime,
+    detectorId,
+    undefined,
+    false
+  );
+
+  // We set the http client in the plugin.ts setup() fn. We pull it in here to make a
+  // server-side call directly.
+  // Note we can't use the redux fns here (e.g., searchResults()) since it requires
+  // hooks (e.g., useDispatch()) which doesn't make sense in this context, plus is not allowed by React.
+  const anomalySummaryResponse = await getClient().post(
+    `..${AD_NODE_API.DETECTOR}/results/_search`,
+    {
+      body: JSON.stringify(anomalySummaryQuery),
+    }
+  );
+
+  return parsePureAnomalies(anomalySummaryResponse);
+};
+
+// This takes anomalies and returns them as vis layer of type PointInTimeEvents
+const convertAnomaliesToLayer = (
+  anomalies: AnomalyData[],
+  ADPluginResource: PluginResource
+): PointInTimeEventsVisLayer => {
+  const events = anomalies.map((anomaly: AnomalyData) => {
+    return {
+      timestamp: anomaly.startTime + (anomaly.endTime - anomaly.startTime) / 2,
+      metadata: {},
+    };
+  });
+  return {
+    originPlugin: 'anomaly-detection',
+    type: VisLayerTypes.PointInTimeEvents,
+    pluginResource: ADPluginResource,
+    events: events,
+  } as PointInTimeEventsVisLayer;
+};
+
+export const overlayAnomaliesFunction =
+  (): OverlayAnomaliesExpressionFunctionDefinition => ({
+    name,
+    type: 'vis_layers',
+    inputTypes: ['vis_layers'],
+    help: i18n.translate('data.functions.overlay_anomalies.help', {
+      defaultMessage: 'Add an anomaly vis layer',
+    }),
+    args: {
+      detectorId: {
+        types: ['string'],
+        default: '""',
+        help: '',
+      },
+    },
+
+    async fn(input, args, context): Promise<ExprVisLayers> {
+      // Parsing all of the args & input
+      const detectorId = get(args, 'detectorId', '');
+      const timeRange = get(
+        context,
+        'searchContext.timeRange',
+        ''
+      ) as TimeRange;
+      const origVisLayers = get(input, 'layers', [] as VisLayers) as VisLayers;
+      const parsedTimeRange = timeRange ? calculateBounds(timeRange) : null;
+      const startTimeInMillis = parsedTimeRange?.min?.unix()
+        ? parsedTimeRange?.min?.unix() * 1000
+        : undefined;
+      const endTimeInMillis = parsedTimeRange?.max?.unix()
+        ? parsedTimeRange?.max?.unix() * 1000
+        : undefined;
+      const ADPluginResource = {
+        type: 'anomaly-detection-type',
+        id: detectorId,
+        name: PLUGIN_NAME,
+        urlPath: `${PLUGIN_NAME}#/detectors`,
+      };
+      try {
+        if (startTimeInMillis === undefined || endTimeInMillis === undefined) {
+          throw new RangeError('start or end time invalid');
+        }
+        const anomalies: AnomalyData[] = await getAnomalies(
```

Comment:
nit: duplicate type declaration not needed since it's defined in `getAnomalies()`